algosdk/template.py

```diff
@@ -175,6 +175,115 @@ def get_program(self):
+            self.close_remainder_address = bytes(constants.address_len)
+        self.receiver = receiver
+        
+    def get_program(self):
+        """
+        Return a byte array to be used in LogicSig.
+        """
+        orig = ("ASAFAgEFBgcmAyD+vKC7FEpaTqe0OKRoGsgObKEFvLYH/FZTJclWlfaiEy" +
+                "DmmpYeby1feshmB5JlUr6YI17TM2PKiJGLuck4qRW2+QEGMgQiEjMAECMS" +
+                "EDMABzEAEhAzAAgxARIQMRYjEhAxECMSEDEHKBIQMQkpEhAxCCQSEDECJR" +
+                "IQMQQhBBIQMQYqEhA=")
+        orig = base64.b64decode(orig)
+        offsets = [5, 6, 7, 11, 44, 76]
+        values = [self.amount, self.first_valid, self.last_valid,
+                  self.receiver, self.close_remainder_address,
+                  base64.b64encode(self.lease_value)]
+        types = [int, int, int, "address", "address", "base64"]
+        return inject(orig, offsets, values, types)
+
+    @staticmethod
+    def get_transactions(txn, lsig, private_key, fee, first_valid, last_valid):
+        """
+        Create and sign the secondary dynamic fee transaction, update
+        transaction fields, and sign as the fee payer; return both
+        transactions.
+
+        Args:
+            txn (Transaction): main transaction from payer
+            lsig (LogicSig): signed logic received from payer
+            private_key (str): the secret key of the account that pays the fee
+                in base64
+            fee (int): fee per byte, for both transactions
+            first_valid (int): first valid round for the transaction
+            last_valid (int): last valid round for the transaction
+        """
+        txn.fv = first_valid
+        txn.lv = last_valid
+        txn.fee = fee
+        txn.fee = max(constants.min_txn_fee, fee*txn.estimate_size())
+
+        # reimbursement transaction
+        address = account.address_from_private_key(private_key)
+        txn_2 = transaction.PaymentTxn(address, fee, first_valid, last_valid,
+                                       txn.genesis_hash, txn.sender, txn.fee,
+                                       lease=txn.lease)
+
+        transaction.assign_group_id([txn, txn_2])
+
+        stx_1 = transaction.LogicSigTransaction(txn, lsig)
+        stx_2 = txn_2.sign(private_key)
+
+        return [stx_1, stx_2]
+
+    def sign_dynamic_fee(self, private_key, gh):
+        """
+        Return the main transaction and signed logic needed to complete the
+        transfer. These should be sent to the fee payer, who can use
+        get_transactions() to update fields and create the auxiliary
+        transaction.
+
+        Args:
+            private_key (bytes): the secret key to sign the contract in base64
+            gh (str): genesis hash, in base64
+        """
+        sender = account.address_from_private_key(private_key)
+
+        # main transaction
+        close = None if self.close_remainder_address == bytes(
+            constants.address_len) else self.close_remainder_address
```

Comment:
I think that `self.close_remainder_address` is always initialized, so you shouldn't need this.